web/src/fileview/fileview.js

```diff
@@ -143,6 +153,22 @@ function init(initData) {
     return true;
   }
 
+  function showFastForwardError() {
```

Comment:
Yep, that should already be in the PR. `Fastforwardlink` in `server/fileview.go` is now set iff 1) the commit is historical AND 2) the file is a proper file, not a directory.

Reply:
I think one last tweak might be necessary? In the livegrep repository history, if I start here:

http://localhost:8910/view/livegrep/server/query_test.go?commit=3d063a6d653e0401#L101

and then press `f` then I get taken to:

http://localhost:8910/view/livegrep/server/query_test.go?commit=847aef0b8a043dda&ffl=112

1. Why `ffl=112` instead of simply `#L112`? There would be great value in not adding another URL parameter, if we could manage it.
2. At this point the `f` option is again offered, but the developer is now stuck in a loop — `f` just takes me back to the same URL.

So it looks like the logic isn't completely working yet for hiding `f` once the hash of the commit being viewed is the current / most-recent hash. If we can just get that working then this will be ready to merge!